exporters/metric/cortex/cortex.go

```diff
@@ -62,3 +68,74 @@ func InstallNewPipeline(config Config, options ...push.Option) (*push.Controller
 	global.SetMeterProvider(pusher.Provider())
 	return pusher, nil
 }
+
+// AddHeaders adds required headers as well as all headers in Header map to a http request.
+func (e *Exporter) addHeaders(req *http.Request) {
+	// Cortex expects Snappy-compressed protobuf messages. These two headers are hard-coded as they
+	// should be on every request.
+	req.Header.Add("X-Prometheus-Remote-Write-Version", "0.1.0")
+	req.Header.Add("Content-Encoding", "snappy")
+	req.Header.Set("Content-Type", "application/x-protobuf")
+
+	// Add all user-supplied headers to the request.
+	for name, field := range e.config.Headers {
+		req.Header.Add(name, field)
+	}
+}
+
+// BuildRequest creates an http POST request with a []byte as the body and headers attached.
+func (e *Exporter) buildRequest(message []byte) (*http.Request, error) {
+	// Create the request with the endpoint and message. The message should be a Snappy-compressed
+	// protobuf message.
+	req, err := http.NewRequest(http.MethodPost, e.config.Endpoint, bytes.NewBuffer(message))
+	if err != nil {
+		return nil, err
+	}
+
+	// Add the required headers and the headers from Config.Headers.
+	e.addHeaders(req)
+
+	return req, nil
+}
+
+// BuildMessage creates a Snappy-compressed protobuf message from a slice of TimeSeries.
+func (e *Exporter) buildMessage(timeseries []*prompb.TimeSeries) ([]byte, error) {
+	// Wrap the TimeSeries as a WriteRequest since Cortex requires it.
+	writeRequest := &prompb.WriteRequest{
+		Timeseries: timeseries,
+	}
+
+	// Convert the struct to a slice of bytes.
+	message, err := proto.Marshal(writeRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	// Compress the message.
+	compressed := snappy.Encode(nil, message)
+
+	return compressed, nil
+}
+
+// ErrSendRequestFailure is an error for when the response does not return status code 200.
+var ErrSendRequestFailure = fmt.Errorf("Failed to send the HTTP request")
+
+// SendRequest sends an http request using the Exporter's http Client. It will not handle retry
+// logic as retrying can more harmful than helpful
+func (e *Exporter) sendRequest(req *http.Request) (int, error) {
```

Comment:
Does this method need to return an `int`, or just an error? What will calling code do with the returned status code?

Reply:
I added it for testing since I had trouble comparing errors when I used `fmt.Errorf()`. I'll change it to return only an error.